### PR TITLE
Clarifies profiles in the documentation

### PR DESCRIPTION
**Pull request by matsduf**
## Purpose

Clarifies the limitations on profiles in the configuration and updates the documentation in the API documentation.

## How to test this PR

Documentation only, no change of implementation.

### CODE REVIEW

File: docs/API.md [hunk 1]
```diff
@@ -227,19 +227,37 @@ The drawback of this setup will be that the GUI will have to wait for at least o
...
+
+When a method receives a legal but undefined *profile name* value for a parameter
+with this type, it returns the following error message:
+
+```json
+{
+    "jsonrpc":"2.0",
+    "id":1,
+    "error":{
+        "message":"Internal error 009 \n",
```
[the remark this answers is not in the document]
Comment (matsduf) [reply docs/API.md:263]: I have updated.

File: docs/API.md [hunk 2]
```diff
@@ -227,19 +227,37 @@ The drawback of this setup will be that the GUI will have to wait for at least o
...
+    "error":{
+        "message":"Invalid method parameter(s).",
+        "data":"/profile: String does not match (?^ui:^[a-z0-9]$|^[a-z0-9][a-z0-9_-]{0,30}[a-z0-9]$).",
+        "code":"-32602"
+    }
+}
+```
+
+When a method receives a legal but undefined *profile name* value for a parameter
+with this type, it returns the following error message:
```
[the remark this answers is not in the document]
Comment (matsduf) [reply docs/API.md:256]: I have added such a text.

**Comment by matsduf** [comment]
@mattias-p, please re-review.

**Review by mattias-p — APPROVED** [review]

**Comment by matsduf** [comment]
@blacksponge, can you review this?

**Review by blacksponge — APPROVED** [review]
looks good to me

**Pull request merged**